tools/codesign_mac_executable.py

```diff
@@ -0,0 +1,232 @@
+#
+# Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+#
+
+# Please refer to https://medium.com/@parthnaik92/codesign-and-notarize-unreal-engine-mac-builds-for-distribution-outside-of-the-mac-store-d2f6e444f3a7
+# for pre-requisites and setup your system before trying to run this file.
+
+import argparse
+import os
+import shutil
+import subprocess
+import time
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--macos_executable", required=True)
+    parser.add_argument("--entitlements_file", required=True)
+    parser.add_argument("--developer_id", required=True)
+    parser.add_argument("--output_dir", required=True)
+    parser.add_argument("--apple_username", required=True)
+    parser.add_argument("--apple_password", required=True)
+    parser.add_argument("--wait_time_seconds", type=float, default=600.0)
+    args = parser.parse_args()
+
+    assert os.path.exists(args.macos_executable)
+    radio_effect_units = [
+        os.path.join(args.macos_executable, "Contents", "Resources", "RadioEffectUnit.component"),
+        os.path.join(args.macos_executable, "Contents", "UE4", "Engine", "Build", "Mac", "RadioEffectUnit")]
+
+    print(f"[SPEAR | codesign_mac_executable.py] Removing {radio_effect_units[0]}.")
+    shutil.rmtree(radio_effect_units[0], ignore_errors=True)
+
+    print(f"[SPEAR | codesign_mac_executable.py] Removing {radio_effect_units[1]}.")
+    shutil.rmtree(radio_effect_units[1], ignore_errors=True)
+
+    print(f"[SPEAR | codesign_mac_executable.py] Changing rpaths...")
+
+    # change current working directory to add relative rpaths 
+    cwd = os.getcwd()
+    os.chdir(os.path.join(args.macos_executable, ".."))
```

Comment:
- create a local variable for macos_executable = os.path.realpath(...)
- get realpath for os.path.join(args.macos_executable, "..")
- print(f"[SPEAR | ...] Changing working directory: {...}") it's worth printing that the script is doing this as a debugging tool, in case I need to execute the various cmd manually myself

Reply:
Done